profiles/views.py

```diff
@@ -194,8 +197,14 @@ def get_serializer_class(self):
     def perform_destroy(self, instance):
         serializer = self.get_serializer(data=self.request.data)
         serializer.is_valid(raise_exception=True)
-        instance.is_deleted = True
-        instance.save()
+        user = self.request.user
+        with transaction.atomic():
+            instance.is_deleted = True
+            instance.save()
+            user.is_active = False
+            user.email = f"is_deleted_{now()}_{user.email}"
+            user.save()
+            djoser_utils.logout_user(self.request)
```

Comment:
I don't know anything about Djoser, but — do we need it here? Can't we do just `logout(self.request)`?

Reply:
The authorization token must be removed from db during logout.
logout(self.request) doesn't do it, and djoser_utils.logout_user(self.request) does.